extensions/bitwarden/src/components.tsx

```diff
@@ -25,32 +39,68 @@ export function TroubleshootingGuide(): JSX.Element {
 }
 
 export function UnlockForm(props: { onUnlock: (token: string) => void; bitwardenApi: Bitwarden }): JSX.Element {
+  const serverUrlPreference = getServerUrlPreference();
   const { bitwardenApi, onUnlock } = props;
-  const [vaultStatus, setVaultStatus] = useState("...");
+  const [vaultState, setVaultState] = useState<VaultState | null>(null);
+  const [isLoading, setLoading] = useState<boolean>(false);
 
   useEffect(() => {
     bitwardenApi.status().then((vaultState) => {
-      if (vaultState.status == "unauthenticated") {
-        setVaultStatus("Logged out");
-      } else {
-        setVaultStatus(`Locked (${vaultState.userEmail})`);
-      }
+      setVaultState(vaultState);
     });
   }, []);
 
+  let userMessage = "...";
+  let serverMessage = "...";
+  if (vaultState) {
```

Comment:
Can we move this block to a separate function ?

Reply:
Good idea, I'll move this into another component.

Reply:
I've refactored this into a hook.

Reply:
Thanks !